### DISCUSSION TITLE
Multiple parent_id on the same  modal. Is this possible?

### DISCUSSION THREAD

**Discussion by MGeurts** [staudenmeir/laravel-adjacency-list#202] [Q&A]
Hi,

I have a **User** model like this:

```
            $table->id();
            
            $table->string('name')->nullable();
            
            $table->unsignedBigInteger('father_id')->nullable();
            $table->unsignedBigInteger('mother_id')->nullable();
```

Can this package be used to manage both self-referencing relationships **father_id** and **mother_id** at the same time?
How do I do this?

Thanks in advance.

**Comment by staudenmeir** [comment]
Hi @MGeurts,
The package does support this kind of relationship but only with a separate pivot table for the connections:
https://github.com/staudenmeir/laravel-adjacency-list#graphs-multiple-parents-per-node-many-to-many

You would need to create a new table (e.g. `relationships`) with the columns `child_id` and `parent_id` and then add two rows for every user – one for the child-mother and one for the child-father relationship. You can store the relationship type (`mother` or `father`) as a [pivot column](https://github.com/staudenmeir/laravel-adjacency-list#graphs-pivot-columns).

**Reply by MGeurts** [reply]
Hi Jonas,

I'm writing a Genealogy application. See [https://genealogy.kreaweb.be/users/101](https://genealogy.kreaweb.be/users/101)

I already tried your **Graphs** package to establish a Family Tree structure. 
But in that case, I'm missing 2 functionalities:

- siblings() to get a collection of brothers and sisters, probably not possible because a user has multiple parents (1 father and 1 mother)
- toTree() to get a hierarchical list of all descendants in 1 query, to display the descendants in a top-down tree

My current solution (not using your package yet) works well, but recursively displaying the family tree (recursive user with children) has the (N+1) query problem.

Thanks for the help.
Marc

**Reply by staudenmeir** [reply]
> siblings() to get a collection of brothers and sisters, probably not possible because a user has multiple parents (1 father and 1 mother)

Do want to get only "real" siblings or also half-siblings?

> toTree() to get a hierarchical list of all descendants in 1 query, to display the descendants in a top-down tree

I have looked into this in the past, but it isn't as straight-forward as with tree relationships due to issues like recursion.

**Reply by staudenmeir** [reply]
`v1.16` added a `toTree()` method for graphs:
https://github.com/staudenmeir/laravel-adjacency-list?tab=readme-ov-file#graphs-nested-results

**Reply by MGeurts** [reply]
Very nice. Thanks Jonas.
But in the meantime I implemented my own solution showing ancestors, descendants, siblings (real, half and plus) using Recursive Common Table Expressions. All without the N+1 problem. But thank you for the effort. I learned a lot by diving into the code.
